MantidQt/SliceViewer/src/SliceViewer.cpp

```diff
@@ -743,7 +755,7 @@ void SliceViewer::setWorkspace(Mantid::API::IMDWorkspace_sptr ws) {
   this->updateDimensionSliceWidgets();
 
   // Only autoscale color bar if box is checked
-  if (m_colorBar->getAutoScale()) {
+  if (!m_firstWorkspaceOpen || m_colorBar->getAutoScale()) {
```

Comment:
Can we re-write this slightly to make it easier to read. How about a non-member function something like `bool shouldAutoScale(isFirstWorkspaceOpen, isAutoscalingOnLoad)` ?